File: projet/projet licence/Projet L3/reflecteurJS/historique.js
const WebSocket = require('ws');
const fs = require('fs');

function main() {
  const wsUrl = 'ws://localhost:3000';
  const ws = new WebSocket(wsUrl);
  const historiqueFichier = "reçu.txt";

  ws.on('open', () => {
    console.log('Connecté au réflecteur.');
  });

  ws.on('message', (message) => {
    console.log(`Message reçu : ${message}`);
    fs.appendFile(historiqueFichier, message + '\n', (err) => {
      if (err) {
        console.error('Erreur lors de l\'écriture dans le fichier :', err);
      }
    });
  });

  ws.on('error', (error) => {
    console.error(`Erreur : ${error}`);
  });

  ws.on('close', () => {
    console.log('Connexion fermée');
    fs.writeFile(historiqueFichier, "", (err) => {
      if (err){
        console.error("Erreur lors de l'effacement du fichier :", err);
      }
    });
  });
}

main();

File: projet/projet licence/Projet L3/reflecteurPY/historique.py
import websocket


def on_message(ws, message):
    print(f"Message reçu : {message}")
    with open("reçu.txt", "a") as fichier:
        fichier.write(message + "\n")

def on_error(ws, error):
    print(f"Erreur : {error}")

def on_close(ws, close_status_code, close_msg):

    print("Connexion fermée")
    with open(HISTORIQUE_FICHIER, "w") as fichier:
        fichier.write("")

def on_open(ws):
    print("Connecté au réflecteur.")

if __name__ == "__main__":
    ws_url = "ws://localhost:3000"
    
    ws = websocket.WebSocketApp(
        ws_url,
        on_message=on_message,
        on_error=on_error,
        on_close=on_close
    )
    ws.on_open = on_open
    ws.run_forever()
